src/core/ViewabilityTracker.ts

```diff
@@ -317,18 +354,43 @@ export default class ViewabilityTracker {
 
     //TODO:Talha optimize this
     private _diffUpdateOriginalIndexesAndRaiseEvents(newVisibleItems: number[], newEngagedItems: number[]): void {
-        this._diffArraysAndCallFunc(newVisibleItems, this._visibleIndexes, this.onVisibleRowsChanged);
+        const minimumViewTime = this._viewabilityConfig && this._viewabilityConfig.minimumViewTime
+        ? this._viewabilityConfig.minimumViewTime
+        : 0;
+        this._diffArraysAndCallFunc(newVisibleItems, this._visibleIndexes, this.onVisibleRowsChanged, minimumViewTime);
         this._diffArraysAndCallFunc(newEngagedItems, this._engagedIndexes, this.onEngagedRowsChanged);
         this._visibleIndexes = newVisibleItems;
         this._engagedIndexes = newEngagedItems;
     }
 
-    private _diffArraysAndCallFunc(newItems: number[], oldItems: number[], func: TOnItemStatusChanged | null): void {
+    private checkMinimumViewTime = (all: number[], now: number[], notNow: number[], minimumViewTime: number, callbackFunc: TOnItemStatusChanged): void => {
+        const that = this;
+        const timeoutId = setTimeout(() => {
+            that.timers.delete(timeoutId);
+
+            const currAll = all.filter((index) => that._visibleIndexes.indexOf(index) >= 0);
```

Comment:
What is the max array batch size of all: number[]

Reply:
`all` has all the list items that are visible in viewport at a time
So it depends on list items sizes